Tensile/Activation.py

```diff
@@ -0,0 +1,800 @@
+    elif cDataType.isSingle():
+      kStr += self.inst("v_and_b32", self.getVgprStr(vgprIdx), "0x7fffffff", self.getVgprStr(vgprIdx), "Remove sign bit")
+    elif cDataType.isDouble():
+      kStr += self.inst("v_and_b32", self.getVgprStr(vgprIdx+1), "0x7fffffff", self.getVgprStr(vgprIdx+1), "Remove sign bit")
+    elif cDataType.isInt32():
+      vgprtemp, vgprPInit = self.vgprActivationPool.registerTemp(1, "vgprtmp_absi32")
+      kStr += self.inst("v_sub_i32", vgpr(vgprtemp), 0, self.getVgprStr(vgprIdx), "x2 = -x")
+      kStr += self.inst("v_max_i32", self.getVgprStr(vgprIdx), vgpr(vgprtemp), self.getVgprStr(vgprIdx), "y = max(x, x2)")
+    else:
+      raise RuntimeError("Unsupported data type %s."%cDataType.toDevice("HIP"))
+    return kStr
+  # Internal Use
+  def getClippedReluAssembly(self, cDataType, vgprIdx, activationAlpha, activationBeta):
+    kStr = ""
+    if cDataType.isHalf():
+      for i in range(0, 2):
+        instStr = "%s src0_sel:WORD_%d src1_sel:WORD_0"%(self.getSgprStr(activationAlpha), i)
+        kStr += self.inst("v_cmp_ge_f16", self.vcc, self.getVgprStr(vgprIdx), instStr, "x > alpha ?")
+        vcc = self.vcc + " dst_sel:WORD_%d dst_unused:UNUSED_PRESERVE src0_sel:WORD_%d src1_sel:WORD_%d"%(i, i, i)
+        kStr += self.inst("v_cndmask_b32", self.getVgprStr(vgprIdx), 0.0, self.getVgprStr(vgprIdx), vcc, "set x to 0 if < alpha")
+      kStr += self.inst("v_pk_min_f16", self.getVgprStr(vgprIdx), self.getSgprStr(activationBeta), self.getVgprStr(vgprIdx), "min(x, beta)")
+    elif cDataType.isSingle():
+      kStr += self.inst("v_cmp_ge_f32", self.vcc, self.getVgprStr(vgprIdx), self.getSgprStr(activationAlpha), "x >= alpha ?")
+      kStr += self.inst("v_cndmask_b32", self.getVgprStr(vgprIdx), 0.0, self.getVgprStr(vgprIdx), self.vcc, "set x to 0 if < alpha")
+      kStr += self.inst("v_min_f32", self.getVgprStr(vgprIdx), self.getSgprStr(activationBeta), self.getVgprStr(vgprIdx), "min(x, beta)")
+    elif cDataType.isDouble():
+      kStr += self.inst("v_cmp_ge_f64", self.vcc, self.getVgprStr(vgprIdx, 2), self.getSgprStr(activationAlpha, 2), "x >= alpha ?")
+      kStr += self.inst("v_cndmask_b32", self.getVgprStr(vgprIdx), 0, self.getVgprStr(vgprIdx), self.vcc, "set x to 0 if < 0")
+      kStr += self.inst("v_cndmask_b32", self.getVgprStr(vgprIdx+1), 0, self.getVgprStr(vgprIdx+1), self.vcc, "set x to 0 if < 0")
+      kStr += self.inst("v_min_f64", self.getVgprStr(vgprIdx, 2), self.getSgprStr(activationBeta, 2), self.getVgprStr(vgprIdx, 2), "min(x, beta)")
+    elif cDataType.isInt32():
+      kStr += self.inst("v_cmp_ge_i32", self.vcc, self.getVgprStr(vgprIdx), self.getSgprStr(activationAlpha), "x >= alpha ?")
+      kStr += self.inst("v_cndmask_b32", self.getVgprStr(vgprIdx), 0.0, self.getVgprStr(vgprIdx), self.vcc, "set x to 0 if < alpha")
+      kStr += self.inst("v_min_i32", self.getVgprStr(vgprIdx), self.getSgprStr(activationBeta), self.getVgprStr(vgprIdx), "min(x, beta)")
+    return kStr
+  # Internal Use
```

Comment:
This way would be more straightforward and easy for reading.
> import math
> import struct
> 
> def double_to_hex(f):
>     return hex(struct.unpack('<Q', struct.pack('<d', f))[0])
> def float_to_hex(f):
>     return hex(struct.unpack('<I', struct.pack('<f', f))[0])
> def f16_to_hex(f):
>     return hex(struct.unpack('<H', struct.pack('<e', f))[0])
>     
> inv_ln2=1/math.log(2)
> print("inverse ln2 ", inv_ln2)
> print(double_to_hex(inv_ln2))
> print(float_to_hex(inv_ln2))
> print(f16_to_hex(inv_ln2))
> print()
> 
> neg_inv_ln2=-1*inv_ln2
> print("negative inverse ln2 ", neg_inv_ln2)
> print(double_to_hex(neg_inv_ln2))
> print(float_to_hex(neg_inv_ln2))
> print(f16_to_hex(neg_inv_ln2))

**Result**
> inverse ln2  1.4426950408889634
> 0x3ff71547652b82fe
> 0x3fb8aa3b
> 0x3dc5
> 
> negative inverse ln2  -1.4426950408889634
> 0xbff71547652b82fe
> 0xbfb8aa3b
> 0xbdc5